firedrake/adjoint/blocks.py

```diff
@@ -55,7 +55,19 @@ def solve_init_params(self, args, kwargs, varform):
         self.adj_args = self.forward_args
 
     if len(self.adj_kwargs) <= 0:
-        self.adj_kwargs = self.forward_kwargs.copy()
+        la_kwargs = [
+            "bcs",
+            "solver_parameters",
+            "nullspace",
+            "transpose_nullspace",
+            "near_nullspace",
+            "options_prefix",
+        ]
+        self.adj_kwargs = dict(
+            (key, value)
+            for key, value in self.forward_kwargs.items()
+            if key in la_kwargs
+        )
```

Comment:
Is this the right fix? The correct adjoint should in some sense do something if a `post_jacobian_callback` is specified. Though I don't think we can determine what that is automatically.

Reply:
What is the typical use of these callbacks within the `NonlinearVariationalSolver`? If they are used to implement some type of continuation strategy to ensure convergence, the adjoint solver will not need them as long as it has access to the final state of the variational form.

Reply:
Sometimes they're used to modify the jacobian itself (with some thing that isn't expressible via UFL). But it's a slightly niche use case, so arguably one could just warn.

Reply:
Something like that?